internal/api/handlers/users/handle_get_user.go

```diff
@@ -0,0 +1,35 @@
+package users
+
+import (
+	"net/http"
+
+	"github.com/gin-gonic/gin"
+	"github.com/ndovnar/family-budget-api/internal/api/error"
+	"github.com/ndovnar/family-budget-api/internal/store"
+	"github.com/rs/zerolog/log"
+)
+
+func (u *Users) HandleGetUser(ctx *gin.Context) {
+	id := ctx.Param("id")
+
+	hasAccess := u.authz.IsUserHasAccessToUser(ctx, id)
+	if !hasAccess {
+		ctx.Error(error.NewHttpError(http.StatusForbidden))
+		return
+	}
+
+	user, err := u.store.GetUserByID(ctx, id)
+	if err != nil {
+		log.Error().Err(err).Msg("failed to get user")
+
+		if err == store.ErrNotFound {
+			ctx.Error(error.NewHttpError(http.StatusNotFound))
+		} else {
+			ctx.Error(error.NewHttpError(http.StatusInternalServerError))
+		}
+
+		return
+	}
+
+	ctx.JSON(http.StatusOK, user)
```

Comment:
@tritonJS826 
Is it ok to return "model" or it's better to add extra struct, something like userResponse?

Reply:
Yeah usually you have separate models for database (Repository models - usually autogenerated somehow) and controllers (DTO). Only service layer use both of them.

Reply:
@tritonJS826 what would be the best option to introduce it?
I am thinking about adding something like transactions_response.go in api/handlers/transactions